python/packages/autogen-core/docs/src/user-guide/core-user-guide/design-patterns/concurrent-agents.ipynb

```diff
@@ -0,0 +1,386 @@
+    "        return TaskResponse(task_id=message.task_id, result=f\"Results by {self.id}\")\n",
+    "\n",
+    "\n",
+    "TASK_TOPIC_TYPE = \"task\"\n",
+    "TASK_RESULTS_TOPIC_TYPE = \"task-results\"\n",
+    "\n",
+    "\n",
+    "@type_subscription(topic_type=TASK_TOPIC_TYPE)\n",
+    "class DelegatorAgent(RoutedAgent):\n",
+    "    def __init__(self, description: str, worker_type: str):\n",
+    "        super().__init__(description)\n",
+    "        self.worker_instances = [AgentId(worker_type, f\"{worker_type}-1\"), AgentId(worker_type, f\"{worker_type}-2\")]\n",
+    "\n",
+    "    @message_handler\n",
+    "    async def on_task(self, message: Task, ctx: MessageContext) -> TaskResponse:\n",
+    "        print(f\"Delegator received task {message.task_id}.\")\n",
+    "\n",
+    "        subtask1 = Task(task_id=\"task-part-1\")\n",
+    "        subtask2 = Task(task_id=\"task-part-2\")\n",
+    "\n",
+    "        worker1_result, worker2_result = await asyncio.gather(\n",
+    "            self.send_message(subtask1, self.worker_instances[0]), self.send_message(subtask2, self.worker_instances[1])\n",
+    "        )\n",
+    "\n",
+    "        combined_result = f\"Part 1: {worker1_result.result}, \" f\"Part 2: {worker2_result.result}\"\n",
+    "        task_response = TaskResponse(task_id=message.task_id, result=combined_result)\n",
+    "\n",
+    "        await self.publish_message(task_response, topic_id=TopicId(type=TASK_RESULTS_TOPIC_TYPE, source=\"default\"))\n",
+    "        return task_response"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Delegator received task main-task.\n",
+      "worker/worker-1 starting task task-part-1\n",
+      "worker/worker-2 starting task task-part-2\n",
+      "worker/worker-1 finished task task-part-1\n",
+      "worker/worker-2 finished task task-part-2\n",
+      "Final result: Part 1: Results by worker/worker-1, Part 2: Results by worker/worker-2\n"
+     ]
+    }
+   ],
+   "source": [
+    "runtime = SingleThreadedAgentRuntime()\n",
+    "\n",
+    "await WorkerAgent.register(runtime, \"worker\", lambda: WorkerAgent(\"Worker Agent\"))\n",
+    "await DelegatorAgent.register(runtime, \"delegator\", lambda: DelegatorAgent(\"Delegator Agent\", \"worker\"))\n",
+    "\n",
+    "runtime.start()\n",
+    "\n",
+    "delegator = AgentId(\"delegator\", \"default\")\n",
+    "response = await runtime.send_message(Task(task_id=\"main-task\"), recipient=delegator)\n",
+    "\n",
+    "print(f\"Final result: {response.result}\")\n",
+    "await runtime.stop_when_idle()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Above, we demonstrated that when sending a message from the runtime to a specific agent (delegator), we can expect a response.\n",
```

Comment:
Now as I read this, it feels a little out of place. Perhaps it's better to move the Closure Agent usage to the second scenario on multiple agents and multiple messages. Specifically, on how to collect results from the workers. We can put it inside a subsection called "Collecting Results" for better discoverability.

Reply:
This way the direct message section can focus on direct message API solely without all the subscription stuff, which is really only for publish message.

Reply:
True. Done!